packages/sdk/src/CanonicalBridge.ts

```diff
@@ -159,87 +217,135 @@ class CanonicalBridge extends Base {
+          // Gnosis requires a higher gas limit
+          gasLimit: 500e3,
+          from,
+        })
+      }
+
+      return (l1CanonicalBridge as L1XDaiForeignOmniBridge).populateTransaction.relayTokens(
+        l1CanonicalToken.address,
+        signerAddress,
+        amount,
+        {
+          // Gnosis requires a higher gas limit
+          gasLimit: 500e3,
+          from,
+        }
       )
+    } else if (chain.equals(Chain.Optimism)) {
+      const l2TokenAddress = this.getL2CanonicalTokenAddress(this.tokenSymbol, chain)
       if (!l2TokenAddress) {
-        throw new Error(
-          `token "${this.tokenSymbol}" on chain "${chain.slug}" is unsupported`
+        throw new Error(`token "${this.tokenSymbol}" on chain "${chain.slug}" is unsupported`)
+      }
+
+      if (this.tokenSymbol === Token.DAI) {
+        const bridge = L1OptimismDaiTokenBridge__factory.connect(
+          l1CanonicalBridge.address,
+          provider
+        )
+        return bridge.populateTransaction.depositERC20(
+          l1CanonicalToken.address,
+          l2TokenAddress,
+          amount,
+          BigNumber.from('2000000'),
+          '0x',
+          {
+            from,
+            gasLimit: 500e3,
+          }
         )
       }
-      const bridge = await this.getContract(
-        L1OptimismTokenBridge__factory,
-        bridgeAddress,
-        provider
-      )
-      await this.checkMaxTokensAllowed(chain, bridge, amount)
-      return bridge.deposit(tokenAddress, l2TokenAddress, recipient, amount)
-    } else if ((chain as Chain).equals(Chain.Arbitrum)) {
-      const arbChain = this.getArbChainAddress(this.tokenSymbol, chain)
-      const bridge = await this.getContract(
-        ArbitrumGlobalInbox__factory,
-        bridgeAddress,
-        provider
-      )
-      await this.checkMaxTokensAllowed(chain, bridge, amount)
-      return bridge.depositERC20Message(
-        arbChain,
-        tokenAddress,
-        recipient,
-        amount
+
+      const bridge = L1OptimismTokenBridge__factory.connect(l1CanonicalBridge.address, provider)
+      return bridge.populateTransaction.deposit(
+        l1CanonicalToken.address,
+        l2TokenAddress,
+        signerAddress,
+        payload,
+        { from, gasLimit: 500e3 }
       )
-    } else if ((chain as Chain).equals(Chain.Polygon)) {
-      const bridgeAddress = this.getL1PosRootChainManagerAddress(
-        this.tokenSymbol,
-        chain
+    } else if (chain.equals(Chain.Arbitrum)) {
+      let bridge: ArbitrumInbox | L1ArbitrumDaiGateway = L1ArbitrumDaiGateway__factory.connect(l1CanonicalBridge.address, provider)
+      if (this.tokenSymbol === Token.ETH) {
+        bridge = await this.getContract(ArbitrumInbox__factory, l1CanonicalBridge.address, provider)
+        return (bridge as ArbitrumInbox).populateTransaction.depositEth(payload, {
+          value: amount,
+          from,
+        })
+      }
+
+      const maxGas = 200e3
+      const gasPriceBid = '100'
```

Comment:
i think these values were obtained from a tx that i populated via the arbitrum bridge. i'm unsure if they are variable. in any case, i'm going to change the values to use the ones found in [this example](https://github.com/OffchainLabs/arbitrum/blob/develop/packages/arb-ts/integration_test/arb-bridge.test.ts#L455-L462), as this is what is recommended by the arb-ts package

Reply:
Have you tested a transaction with these values on mainnet? I know their tests have these values, but their L2 recipient or alias address might be pre-funded for these tests, which means it would work in the test but not in prod.

From a top level perspective, the user needs to pay for the bridging of the asset when using the Arbitrum bridge. This could come in the form of having funds pre-deposited on an address in L2 OR sending ETH with the L1 tx. In this case, if the `gasPriceBid` is `0`, then I think we are telling the system that we expect the funds to already exist on L2 to be paid, which I do not believe will be the case for Hop users.

It would be best to test this on the actual Arbitrum bridge itself and trace the tx and see what happens, as some of my info above might be outdated or misinformed.

Happy to call and talk through this if that is easier.

Reply:
wow, it really varies. here's two successful deposits using the arbitrum bridge ui:

![Screen Shot 2022-03-07 at 9 40 25 AM](https://user-images.githubusercontent.com/18040654/157055710-743bc6bf-d3ab-4d74-b8b5-ae4ed5882f27.png)

![Screen Shot 2022-03-07 at 9 40 22 AM](https://user-images.githubusercontent.com/18040654/157055737-f410aed7-da8c-4d9f-910f-f4b6b846b834.png)

Reply:
it may be best to integrate `arb-ts` (https://developer.offchainlabs.com/docs/frontend_integration#arbitrum--ethereum-bridge). it seems pretty straightforward